File: my_py_pkg/my_py_pkg/robot_news_publisher.py
import rclpy
from rclpy.node import Node

from std_msgs.msg import String


class RobotNewsPublisher(Node):
    def __init__(self):
        super().__init__('robot_news_publisher')
        self.declare_parameter('robot_name', 'X001')
        self.declare_parameter('publish_freq', 2)

        self.robot_name = self.get_parameter('robot_name').value
        publish_freq = self.get_parameter('publish_freq').value


        self.publisher_ = self.create_publisher(String, 'robot_news', 10)
        self.timer_ = self.create_timer(1/publish_freq, self.publish_news)

        self.get_logger().info('Started robot_news publisher')

    def publish_news(self):
        msg = String()
        msg.data = f"hello, {self.robot_name} is ready to take over!"

        self.publisher_.publish(msg)


def main(args=None):
    rclpy.init(args=args)

    node = RobotNewsPublisher()
    rclpy.spin(node)

    rclpy.shutdown()


if __name__ == '__main__':
    main()

File: my_cpp_pkg/src/robot_news_publisher.cpp
#include "rclcpp/rclcpp.hpp"
#include "std_msgs/msg/string.hpp"


using namespace std::chrono_literals;

class RobotNewsPublisher : public rclcpp::Node {
    public:
    RobotNewsPublisher() : Node("robot_news_publisher") {
        publisher_ = this->create_publisher<std_msgs::msg::String>("robot_news", 10);
        timer_ = this->create_wall_timer(500ms, std::bind(&RobotNewsPublisher::publishNews, this));

        RCLCPP_INFO(this->get_logger(), "Robot News Publisher CPP has started");
    }

    private:
    void publishNews() {
        auto msg = std_msgs::msg::String();
        msg.data = "Hello world " + robot_name;

        publisher_->publish(msg);
    }
    std::string robot_name = "X001_CPP";
    rclcpp::Publisher<std_msgs::msg::String>::SharedPtr publisher_;
    rclcpp::TimerBase::SharedPtr timer_;
};

int main(int argc, char** argv) {
    rclcpp::init(argc, argv);

    auto node = std::make_shared<RobotNewsPublisher>();
    rclcpp::spin(node);
    rclcpp::shutdown();

    return 0;
}